lib/Models/Definition/addModelStrataView.ts

```diff
@@ -49,3 +49,12 @@ export default function addModelStrataView<
 
   return model;
 }
+
+function decorate(
+  target: any,
+  decorators: { [id: string]: PropertyDecorator }
+) {
```

Comment:
I've had a look into this, and I think this only works because the caller of this (CreateModel) puts `makeObservable` in the constructor.

Reply:
The mobx4/5 decorate utility was [removed in mobx6](https://mobx.js.org/migrating-from-4-or-5.html#upgrading-classes-to-use-makeobservable).

> this only works because the caller of this (CreateModel) puts makeObservable in the constructor

Yeah, as you already know, in mobx6 the role of decorators is [simply to note](https://github.com/mobxjs/mobx/blob/58bb052ca41b8592e5bd5c3003b68ec52da53f33/packages/mobx/src/api/computed.ts#L38-L41) the properties that need to be enhanced. The actual enhancement is done by the call to [makeObservable](https://github.com/mobxjs/mobx/blob/58bb052ca41b8592e5bd5c3003b68ec52da53f33/packages/mobx/src/api/makeObservable.ts#L43) in the constructor. So yes, this wouldn't work without a call to `makeObservable`.